### PR TITLE
Feat-Go: add Trim function

### PR DESCRIPTION
**Pull request by YogiPristiawan**
- Programming langugage: 
- Function(s) that I'm implementing:
- Issue number: 

- [x] I have created unit tests to support my implementation
- [x] I have created in-code documentation (or docstring) to document my code
- [ ] (Optional) I have updated the README.md file to document the function I'm implementing

### CODE REVIEW

File: go/trim.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	reversedString := Strrev(strings.Join(chars, ""))
+	// use regex to get range pattern
+	r := regexp.MustCompile(`.\.{2}.`)
+	// concate non-range char
+	charLists = strings.Join(r.Split(reversedString, -1), "")
+	// range char pattern
+	rangedChar := r.FindAll([]byte(reversedString), -1)
+	// generates correct characters between ranges
+	for _, val := range rangedChar {
+		if val[3] > val[0] {
```
Comment (aldy505) [inline go/trim.go:38]: val[3] bisa kena panic gara-gara index does not exists

Comment (YogiPristiawan) [reply go/trim.go:38]: kayaknya engga pak, soalnya itu dari patternya regex kan

Comment (YogiPristiawan) [reply go/trim.go:38]: kayaknya engga pak, soalnya itu dari patternya regex kan

Comment (YogiPristiawan) [reply go/trim.go:38]: ini gimana pak? done kah?

Comment (aldy505) [reply go/trim.go:38]: ya engga, benerin dulu

**Review by elianiva — APPROVED** [review]

**Pull request merged**